### ISSUE TITLE
[Bug]: Accessibility Bug - Nested Interactive Controls Should not be Nested - Details List View

### ISSUE THREAD

**Issue by avyaktawrat** [microsoft/fluentui#26083]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 9.20 GB / 31.71 GB
  Browsers:
    Edge: Spartan (44.22621.963.0), Chromium (108.0.1462.54)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/avyaktawrat/pen/qByddNm

## Actual Behavior
On adding a checkbox instead of the original radio buttons, we observe an Accessibility Bug - Nested Interactive Control Failure, as shown in the screenshot below.
![image](https://user-images.githubusercontent.com/46568112/209158545-058b50bd-bd86-41de-911f-692b00b1ab36.png)

This is just a sample code for demonstration purposes of Accessibility Bug

## Expected Behavior
Upon replacing the radio buttons with checkboxes, the above error should not be observed.

Please let us know if there is anything else that can be changed to make Details List accessible with checkbox control.

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
This is because `onRenderCheckbox` controls what is rendered inside the checkbox in DetailsList. Rendering a `<Checkbox>` I'd suggest using it to render the specific checkbox icon you want, since that allows us to handle both the selection and the accessible name and other semantics for you. Right now in your codepen for example, the inner checkbox authored in `onRenderCheckbox` does not have a label. Within DetailsList, the outer checkbox falls back on "select row" as a label, but will also incorporate the rowheader for a unique accessible name, if one is defined.

If you really want to fully override the checkbox, you can use `onRenderRow` like so:
```tsx
const onRenderCheckbox = (checkboxProps: any) => (
    <div>
      <Checkbox title = "Test Checkboxes" checked = {this._selection}/>
    </div>
  );

const onRenderRow = (props: IDetailsRowProps) => {
  return <DetailsRow {...props} onRenderCheck={customOnRenderCheck} />;
};

<DetailsList {...props} onRenderRow={onRenderRow} />
```

The above will require redoing some of our internal checkbox logic, including the per-row unique accessible name. For ease, I'd recommend using `onRenderCheckbox` + icons.
